src/Analysis/Engine/Impl/PythonAnalyzer.cs

```diff
@@ -622,7 +622,16 @@ class LazyModuleEnumerator {
 
         public AnalysisLimits Limits {
             get { return _limits; }
-            set { _limits = value; }
+            set {
+                var limits = _limits;
+                _limits = value;
+
+                if ((limits == null && _limits != null)
```

Comment:
Can it be the other way: `limits != null && _limits == null`?

Reply:
Actually I think it should be using default and never be null